spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -237,51 +206,50 @@ def __init__(self, ipyclient, additional_options, interpreter_versions,
         # Show a message in our installers to explain users how to use
         # modules that don't come with them.
         self.show_modules_message = is_pynsist() or running_in_mac_app()
-        self.shutdown_lock = Lock()
 
     # ---- Public API ---------------------------------------------------------
-    def shutdown_kernel(self):
-        """Shutdown kernel."""
-        with self.shutdown_lock:
-            # Avoid calling shutdown_kernel on the same manager twice
-            # from different threads to avoid crash.
-            if self.kernel_manager.shutting_down:
-                return
-            self.kernel_manager.shutting_down = True
-        try:
-            self.kernel_manager.shutdown_kernel()
-        except Exception:
-            # kernel was externally killed
-            pass
+    @property
+    def is_spyder_kernel(self):
+        if self.kernel_handler is None:
```

Comment:
This check will fail if `self.kernel_handler` is `False` in the constructor, for instance.

Reply:
I am not sure I understand how `self.kernel_handler` could be false, shouldn't it either be `None` or a kernel handler instance?

Reply:
Ok I see where I made a mistake